homeassistant/components/wallbox/coordinator.py

```diff
@@ -174,6 +215,55 @@ async def async_set_charging_current(self, charging_current: float) -> None:
         )
         await self.async_request_refresh()
 
+    @_require_authentication
+    def _get_sessions(
+        self,
+        charger_id: str,
+        start_date: datetime.datetime,
+        end_date: datetime.datetime,
+    ) -> list[WallboxEvent]:
+        """Get charging sessions between timestamps for Wallbox."""
+        data = self._wallbox.getSessionList(charger_id, start_date, end_date)
+        tzone = dt_util.get_default_time_zone()
+        events: list[WallboxEvent] = [
+            WallboxEvent(
+                charger_name=event[CHARGER_ATTRIBUTES_KEY][CHARGER_CHARGER_NAME_KEY],
+                username=event[CHARGER_ATTRIBUTES_KEY][CHARGER_USERNAME_KEY],
+                session_id=event["id"],
+                currency=event[CHARGER_ATTRIBUTES_KEY]["currency"]["code"],
+                serial_number=event[CHARGER_ATTRIBUTES_KEY][CHARGER_CHARGER_KEY],
+                time=event[CHARGER_ATTRIBUTES_KEY][CHARGER_TIME_KEY],
+                energy=event[CHARGER_ATTRIBUTES_KEY][CHARGER_MID_ENERGY_KEY]
+                if event[CHARGER_ATTRIBUTES_KEY][CHARGER_MID_ENERGY_KEY] > 0
+                else event[CHARGER_ATTRIBUTES_KEY][CHARGER_ENERGY_KEY],
+                session_cost=event[CHARGER_ATTRIBUTES_KEY][CHARGER_COST_KEY],
+                start=datetime.datetime.fromtimestamp(
+                    event[CHARGER_ATTRIBUTES_KEY][CHARGER_START_KEY], tzone
+                ),
+                end=datetime.datetime.fromtimestamp(
+                    event[CHARGER_ATTRIBUTES_KEY]["end"], tzone
+                ),
+                summary=f"Charging session {event["id"]}: {event[CHARGER_ATTRIBUTES_KEY][CHARGER_ENERGY_KEY]}KWh",
+                location=event[CHARGER_ATTRIBUTES_KEY][CHARGER_CHARGER_NAME_KEY],
+                description=f"Session ID: {event["id"]}\nSerial number: {event[CHARGER_ATTRIBUTES_KEY][CHARGER_CHARGER_KEY]}\nUsername: {event[CHARGER_ATTRIBUTES_KEY][CHARGER_USERNAME_KEY]}\nTime: {datetime.datetime.fromtimestamp(event[CHARGER_ATTRIBUTES_KEY][CHARGER_TIME_KEY]) - datetime.datetime.fromtimestamp(0)}\nEnergy: {event[CHARGER_ATTRIBUTES_KEY][CHARGER_MID_ENERGY_KEY] if event[CHARGER_ATTRIBUTES_KEY][CHARGER_MID_ENERGY_KEY] > 0 else event[CHARGER_ATTRIBUTES_KEY][CHARGER_ENERGY_KEY]}KWh\nSession cost: {event[CHARGER_ATTRIBUTES_KEY][CHARGER_COST_KEY]}{event[CHARGER_ATTRIBUTES_KEY]["currency"]["code"]}\n",
+            )
+            for event in data[CHARGER_SESSION_DATA_KEY]
+            if event[CHARGER_TYPE_KEY] == "charger_log_session"
+        ]
```

Comment:
Do you mean creating functions to fill in the parameters of the wallboxEvent?

Or do you mean creating services to get more info of the event?

Reply:
Hey, I haven't got a reply yet. What do you mean?

Reply:
Ah, so the regular get events only gives back normal events, and create a new service get_wallbox_events that gives it with more info?

I know this might be off topic (or at least not in this commit) but wouldn't it be better if the get events call had an optional parameter: compact?